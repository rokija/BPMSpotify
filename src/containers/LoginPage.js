import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { getAuth } from '../actions/authActions';

let lFollowX = 0,
    lFollowY = 0,
    x = 0,
    y = 0,
    friction = 1 / 20;

export class LoginPage extends Component {
    constructor() {
        super();

        this.logIn = this.logIn.bind(this);
    }

    componentWillMount() {
        const {authReducer } = this.props;
        if (authReducer.isLogged) {
            this.context.router.history.push("/");
        }
        else {
            this.context.router.history.push("/login");
        }
    }

    logIn() {
        this.props.dispatch(getAuth());
        return true;
    }

    mouseOverAnimation(e) {
        function moveBackground() {
            x += (lFollowX - x) * friction;
            y += (lFollowY - y) * friction;
            const background = document.getElementsByClassName("background-image-login-page")[0];
            background.style.transform = 'translate(' + x + 'px, ' + y + 'px) scale(1.2)';
            window.requestAnimationFrame(moveBackground);
        }

        let lMouseX = Math.max(-100, Math.min(100, window.innerWidth / 2 - e.clientX));
        let lMouseY = Math.max(-100, Math.min(100, window.innerHeight / 2 - e.clientY));
        lFollowX = (20 * lMouseX) / 100;
        lFollowY = (10 * lMouseY) / 100;

        if (window.innerWidth > 1118) {
            moveBackground();
            return true;
        }
    }

    render() {
        return (
            <div className="login-page-wrapper">
                <div className="background-image-login-page" onMouseMove={(e) => this.mouseOverAnimation(e)} />
                <div className="login-button-label-wrapper" onMouseMove={(e) => this.mouseOverAnimation(e)}>
                    <h2>Please Log in with</h2>
                    <div className="spotify-icon-black" />
                    <button className="login-button" onClick={() => this.logIn()}>Log in</button>
                </div>
            </div>
        );
    }
}

LoginPage.propTypes = {
    dispatch: PropTypes.func,
    authReducer: PropTypes.object,
    userLoginState: PropTypes.object
};

LoginPage.contextTypes = {
    router: PropTypes.object
};

export function mapStateToProps (state) {
    const {authReducer} = state;
    return {
        authReducer
    };
}
export function mapDispatchToProps (dispatch) {
    return {
        dispatch,
        actions: bindActionCreators({getAuth}, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
