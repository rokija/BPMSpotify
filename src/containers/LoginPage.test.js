import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LoginPage, mapDispatchToProps, mapStateToProps } from './LoginPage';

const context = { router: { history: [] }  };

describe('LoginPage if user is Logged in', () => {
    const minProps = {
        dispatch: jest.fn(),
        authReducer: { isLogged: true }
    };

    const wrapper = mount(<LoginPage {...minProps} />, { context });

    it('renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });
});

describe('LoginPage if user is not Logged in', () => {
    const minProps = {
        dispatch: jest.fn(),
        authReducer: { isLogged: false }
    };

    const wrapper = mount(<LoginPage {...minProps} />, { context });

    it('renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('calls logIn function', () => {
        wrapper.find('button').simulate('click');
    });

    it('calls mapStateToProps', () => {
        expect(mapStateToProps({authReducer: { isLogged: true }})).toEqual({"authReducer": {"isLogged": true}});
    });

    it('calls typeof mapDispatchToProps', () => {
        expect(typeof (mapDispatchToProps())).toEqual("object");
    });
});

describe('LoginPage with shallow wrapper', () => {
    const minProps = {
        dispatch: jest.fn(),
        authReducer: { isLogged: false }
    };

    const wrapper = shallow(<LoginPage {...minProps} />, { context });
    const wrapperInstance = wrapper.instance();

    it('calls logIn', () => {
        expect(wrapperInstance.logIn()).toEqual(true);
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});