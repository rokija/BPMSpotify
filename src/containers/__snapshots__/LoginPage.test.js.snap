// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage with shallow wrapper matches snapshot 1`] = `
<div
  className="login-page-wrapper"
>
  <div
    className="background-image-login-page"
    onMouseMove={[Function]}
  />
  <div
    className="login-button-label-wrapper"
    onMouseMove={[Function]}
  >
    <h2>
      Please Log in with
    </h2>
    <div
      className="spotify-icon-black"
    />
    <button
      className="login-button"
      onClick={[Function]}
    >
      Log in
    </button>
  </div>
</div>
`;
